extensions/amp-playbuzz/0.1/utils.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Returns a function, that, as long as it continues to be invoked, will not
+// be triggered. The function will be called after it stops being called for
+// N milliseconds. If `immediate` is passed, trigger the function on the
+// leading edge, instead of the trailing.
+
+import {dev} from './../../../src/log';
+
+export function debounce(func, wait, immediate) {
+  let timeout;
+  return function() {
+    const context = this, args = arguments;
+    clearTimeout(timeout);
+    timeout = setTimeout(function() {
+      timeout = null;
+      if (!immediate) { func.apply(context, args); };
+    }, wait);
+    if (immediate && !timeout) { func.apply(context, args); }
+  };
+};
+
+export function getElementCreator(document) {
+  return function createElement(name, className, children) {
+    const element = document.createElement(name);
+    element.className = className;
+    appendChildren(element, children);
+    return element;
+  };
+}
+
+function appendChildren(element, children) {
+  children = (!children) ? [] : Array.isArray(children) ? children : [children];
+  children.forEach(child => element.appendChild(child));
+};
+
+export function handleMessageByName(event, messageName, handler) {
+  const isPlaybuzzItemEvent = event.origin &&
+        event.origin.indexOf &&
+        event.origin.indexOf('playbuzz.com') >= 0;
```

Comment:
This has to be a tighter origin check. Also, you know the origin exactly since you set it yourself and even better - you should be able to check the `event.source` directly against your iframe.